### ISSUE TITLE
hashlib.md5(str(salt).encode()).hexdigest() is not FIPS compliant

### ISSUE THREAD

**Issue by jjtroberts** [datafolklabs/cement#626]
# Cement Issue Reporting and Feature Requests

md5 is not FIPS compliant.

Please include any of the following in bug/issue reports where it is relevant:

**System Information**

- Cement Version: 3.0.6
- Python Version: 3.9.13
- Operating System and Version: RHEL8

**Steps to Reproduce (Bugs Only)**

- Enable FIPS on a RHEL8 node
- Execute `make virtualenv`

The following is an [MCVE] (*Minimal Complete
Verifiable Example*) that reproduces the issue:

```python
$ venv/bin/python setup.py develop
Traceback (most recent call last):
  File "/builds/ironbank-tools/robotnik/setup.py", line 2, in <module>
    from robotnik.core.version import get_version
  File "/builds/ironbank-tools/robotnik/robotnik/core/version.py", line 1, in <module>
    from cement.utils.version import get_version as cement_get_version
  File "/builds/ironbank-tools/robotnik/venv/lib/python3.9/site-packages/cement/__init__.py", line 4, in <module>
    from .core.foundation import App, TestApp
  File "/builds/ironbank-tools/robotnik/venv/lib/python3.9/site-packages/cement/core/foundation.py", line 1686, in <module>
    class TestApp(App):
  File "/builds/ironbank-tools/robotnik/venv/lib/python3.9/site-packages/cement/core/foundation.py", line 1696, in TestApp
    class Meta:
  File "/builds/ironbank-tools/robotnik/venv/lib/python3.9/site-packages/cement/core/foundation.py", line 1697, in Meta
    label = "app-%s" % misc.rando()[:12]
  File "/builds/ironbank-tools/robotnik/venv/lib/python3.9/site-packages/cement/utils/misc.py", line 36, in rando
    return hashlib.md5(str(salt).encode()).hexdigest()
ValueError: [digital envelope routines: EVP_DigestInit_ex] disabled for FIPS
```

[MCVE]: https://stackoverflow.com/help/mcve

**Comment by jjtroberts** [comment]
Can [this](https://github.com/datafolklabs/cement/blob/bef815907e3d461c56ca41491d8a5a58020423e2/cement/utils/misc.py#L36) be updated to use `sha256`?

**Comment by derks** [comment]
@jjtroberts this has been merge to `main` and will be included in `stable/3.0.8`. To test, I did the following:

- Built clean CentOS Stream 8 box
- Installed Cement 3.0.6
- Generated a new project
- Verified `make virtualenv` functions with FIPS disabled
- Enabled FIPS
- Verified `make virtualenv` did NOT function with FIPS enabled
- Updated `cement/utils/misc.py` with changes
- Verified `make virtualenv` functions with FIPS enabled
